v3/gateway.proto

```diff
@@ -0,0 +1,173 @@
+    TerrestrialAntennaPlacement antenna_placement = 2;
+    // Number of antennas.
+    // Home Networks receive this value if the Gateway Visibility has antenna_count set.
+    google.protobuf.UInt32Value antenna_count = 3;
+    // Indicates whether the gateway produces fine timestamps.
+    // Home Networks receive this value if the Gateway Visibility has fine_timestamps set.
+    google.protobuf.BoolValue fine_timestamps = 4;
+  }
+  message Satellite {
+    // Satellite location.
+    // Home Networks receive this value if the Gateway Visibility has location set.
+    Location location = 1;
+    // Field of view (meters).
+    // Home Networks receive this value if the Gateway Visibility has location set.
+    double field_of_view = 2;
+    // Trajectory, typically containing the recent past and projected path.
+    // Home Networks receive this value if the Gateway Visibility has location set.
+    repeated TimedLocation trajectory = 3;
+  }
+  oneof value {
+    Terrestrial terrestrial = 1;
+    Satellite satellite = 2;
+  }
+}
+
+message GatewayLocationValue {
+  GatewayLocation location = 1;
+}
+
+message Gateway {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 net_id = 1;
+  // Tenant ID managed by the Forwarder Member.
+  string tenant_id = 14;
+  // Forwarder cluster of the Forwarder Member (optional).
+  string cluster_id = 2;
+  // Identifier of the gateway.
+  GatewayIdentifier gateway_id = 15;
+  // Timestamp when the gateway was last updated.
+  google.protobuf.Timestamp updated_at = 4;
+
+  // Gateway location.
+  GatewayLocation location = 20;
+
+  // Administrative contact.
+  // Home Networks receive this value if the Gateway Visibility has contact_info set.
+  ContactInfo administrative_contact = 16;
+  // Technical contact.
+  // Home Networks receive this value if the Gateway Visibility has contact_info set.
+  ContactInfo technical_contact = 17;
+
+  // Indicates whether the gateway is online.
+  // Home Networks receive this value if the Gateway Visibility has status set.
+  google.protobuf.BoolValue online = 10;
+
+  // Frequency plan of the gateway.
+  // Home Networks receive this value if the Gateway Visibility has frequency_plan set.
+  GatewayFrequencyPlan frequency_plan = 13;
+
+  // Received packets rate (number of packets per hour).
+  // Home Networks receive this value if the Gateway Visibility has packet_rates set.
+  google.protobuf.FloatValue rx_rate = 18;
+  // Transmitted packets rate (number of packets per hour).
+  // Home Networks receive this value if the Gateway Visibility has packet_rates set.
+  google.protobuf.FloatValue tx_rate = 19;
+
+  reserved 3; // Deprecated eui;
+  reserved 5; // Deprecated terrestrial;
+  reserved 6; // Deprecated satellite;
+  reserved 7; // Deprecated contact_name;
+  reserved 8; // Deprecated contact_email;
+  reserved 9; // Deprecated contact_url;
+  reserved 11; // Deprecated rx_count;
+  reserved 12; // Deprecated tx_count;
```

Comment:
This doesn't win the beauty contest in API iterations, but I'm really reluctant to introduce a `v4` API now. Doing that would ideally make it independent from `v3`, which means copying lots of messages and renumbering fields. `Gateway` wasn't being used in any known implementation.